Polyfills/XMLHttpRequest/Source/XMLHttpRequest.cpp

```diff
@@ -106,10 +113,19 @@ namespace Babylon::Polyfills::Internal
 
     Napi::Value XMLHttpRequest::GetResponse(const Napi::CallbackInfo&)
     {
-        const gsl::span<const std::byte> responseBuffer{m_request.ResponseBuffer()};
-        const auto arrayBuffer{Napi::ArrayBuffer::New(Env(), responseBuffer.size())};
-        std::memcpy(arrayBuffer.Data(), responseBuffer.data(), arrayBuffer.ByteLength());
-        return std::move(arrayBuffer);
+        // check response type
+        if (m_request.ResponseType() == UrlLib::UrlResponseType::String)
+        {
+            return Napi::Value::From(Env(), m_request.ResponseString().data());
+        }
+        else
+        {
+            // if it is text, return responseText
+            gsl::span<const std::byte> responseBuffer{ m_request.ResponseBuffer() };
+            auto arrayBuffer{ Napi::ArrayBuffer::New(Env(), responseBuffer.size()) };
```

Comment:
There is an overload that takes an external data pointer. Assuming m_request doesn't get deleted, this should be better than copying.

Reply:
Are you referring to the ` return Napi::Value::From(Env(), m_request.ResponseString().data());` line? If so, I was trying to keep it the same as what was already being done in `XMLHttpRequest::GetResponseText(const Napi::CallbackInfo&)`

Reply:
Discussed offline. Let's copy for now since it's safer.